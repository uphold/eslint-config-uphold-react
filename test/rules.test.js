'use strict';

/**
 * Module dependencies.
 */

const { ESLint } = require('eslint');
const path = require('path');

/**
 * Tests for `eslint-config-uphold-react`.
 */

describe('eslint-config-uphold-react', () => {
  const linter = new ESLint({
    ignore: false,
    overrideConfigFile: path.join(__dirname, '..', 'src', 'index.js')
  });

  test('should not generate any violation for correct code', async () => {
    const source = path.join(__dirname, 'fixtures', 'correct.js');
    const results = await linter.lintFiles([source]);

    expect(results[0].errorCount).toEqual(0);
  });

  it('should generate violations for incorrect code', async () => {
    const source = path.join(__dirname, 'fixtures', 'incorrect.js');
    const results = await linter.lintFiles([source]);

    const rules = results[0].messages.map(violation => violation.ruleId);

    expect(rules).toMatchSnapshot();
  });
});
