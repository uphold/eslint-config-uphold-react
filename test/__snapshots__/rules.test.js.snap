// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-uphold-react should generate violations for incorrect code 1`] = `
[
  "react-hooks/rules-of-hooks",
  "react/prefer-stateless-function",
  "prettier/prettier",
]
`;
